### ISSUE TITLE
PyNEST does not handle NEST error messages with non-ASCII characters correctly

### ISSUE THREAD

**Issue by heplesser** [nest/nest-simulator#406]
PyNEST does not handle error messages containing non-ASCII characters correctly, at least under Python 2.7. 

This is an example of a properly handled error (shortened):

```
In [2]: nest.sli_run('foo')
NESTError                                 Traceback (most recent call last)
[snip]
NESTError: DictError in SLIInterpreter::execute: Key '/foo' does not exist in dictionary.
```

and here an example that breaks: 

```
In [3]: nest.sli_run('føø')
UnicodeDecodeError                        Traceback (most recent call last)

UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 2: ordinal not in range(128)
```

Note that SLI handles this fine:

```
SLI ] føø

Jun 23 15:45:59 SLIInterpreter::execute [Error]: DictError
    Key '/føø' does not exist in dictionary.
```

This issue was originally mentioned in #405.

**Comment by stinebuu** [comment]
PyNEST manages to handle the error correctly with Python 3, we are looking at the problem with Python 2.7 now.

**Comment by heplesser** [comment]
#536 will fix this.

**Comment by hakonsbm** [comment]
@heplesser This issue has been fixed by #536.
